model/src/pyrenew/latent/i0.py

```diff
@@ -80,8 +81,10 @@ def sample(
             Tuple with the initial infections.
         """
         return (
-            npro.sample(
-                name=self.name,
-                fn=self.i0_dist,
+            jnp.atleast_1d(
```

Comment:
is there a reason to enforce this here versus deferring it to the `i0_dist`?

Reply:
If you want i0_dist to be something from `numpyro.distributions`, you have to do this (or see below) to return an array instead of a float. I don't exactly recall why this causes problems, but I know I had to do this in my work as well.

I do handle it slightly differently inhttps://github.com/CDCgov/multisignal-epi-inference/pull/124/:
https://github.com/CDCgov/multisignal-epi-inference/blob/16c1be49eed32b0d4571f8576daab4e5006d01ed/model/src/pyrenew/latent/i0.py#L86

Let's avoid creating a merge conflict by picking a pattern now. Assuming @gvegayon prefers    `jnp.atleast_1d` and I prefer `sample_shape`, @dylanhmorris can break the tie.

Reply:
Unidimensional arrays in `numpyro` get an empty shape, which was causing issues with the convolve functions that needed to index arrays:

![image](https://github.com/CDCgov/multisignal-epi-inference/assets/893619/2022d082-15e4-4513-bf56-88fbec7589d9)

Reply:
I'm with you, @damonbayer, but the issue is that `latent.I0` receives a `dist` during construction, which gives space for the user to sample an `I0` with shape other than `(1,)`. I think `jnp.atleast_1d()` as a reason to exist, and this is the case where it will help more than hurt, IMO.

Reply:
Setting `sample_shape=(1,)` should be the default practice within the code, but here is a bit out of our hands if the user chooses to pass something else (that doesn't have the shape = `(1,)`).

Reply:
Thanks both for the good explanations. Now that I understand the issue, I vote `jnp.atleast_1d()` on the grounds that this is a case where type coercion makes sense. From the user's PoV, a scalar makes conceptual sense, and it's only the computational infrastructure that needs to be told that scalars should always be treated as length-1 vectors.

Reply:
Here is an example:

![image](https://github.com/CDCgov/multisignal-epi-inference/assets/893619/45b0a86c-7ba5-4a1d-b254-c086b18fe3fb)

Here is the code:

```python
import numpyro as npro
import numpyro.distributions as dist
import jax.numpy as jnp

mu = jnp.zeros(2)
with npro.handlers.seed(rng_seed=0):
    x0 = npro.sample('x', dist.Normal(mu, 1))
    x1 = npro.sample('x', dist.Normal(mu, 1), sample_shape=(1,))
    x2 = jnp.atleast_1d(npro.sample('x', dist.Normal(mu, 1)))
print(f'shape of x0: {x0.shape}')
print(f'shape of x1: {x1.shape}')
print(f'shape of x2: {x2.shape}') # Same as x0

print(f'x0: {x0}')
print(f'x1: {x1}')
print(f'x2: {x2}') # Same as x0
```